packages/ui/src/components/GroupSelectorSheet.tsx

```diff
@@ -0,0 +1,165 @@
+import * as db from '@tloncorp/shared/dist/db';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  NativeScrollEvent,
+  NativeSyntheticEvent,
+  SectionListRenderItemInfo,
+} from 'react-native';
+
+import { Stack, View, YStack } from '../core';
+import { AlphaSegmentedGroups } from '../hooks/groupsSorters';
+import { triggerHaptic } from '../utils';
+import { Icon } from './Icon';
+import { GroupListItem, ListItem } from './ListItem';
+import { BlockSectionList } from './SectionList';
+import { Sheet } from './Sheet';
+
+interface Props {
+  onSelect?: (group: db.Group) => void;
+  multiSelect?: boolean;
+  selected?: string[];
+  onScrollChange?: (scrolling: boolean) => void;
+  alphaSegmentedGroups: AlphaSegmentedGroups;
+}
+
+export function GroupSelector(props: Props) {
```

Comment:
This component duplicates a lot of aspects of the AddDmSheet -- are there common functionalities or components we could pull out?

Reply:
extracting the scroll listeners into a hook didn't work as I expected it would. Leaving be for now, but filed an issue here (TLON-2501)